src/sync.ts

```diff
@@ -1911,52 +1923,46 @@ export function _createAndReEmitRoom(client: MatrixClient, roomId: string, opts:
  *
  * Converts the events into `MatrixEvent`s, and emits appropriate {@link ClientEvent.ToDeviceEvent} events.
  * */
-export function processToDeviceMessages(toDeviceMessages: IToDeviceEvent[], client: MatrixClient): void {
+export function processToDeviceMessages(toDeviceMessages: ReceivedToDeviceMessage[], client: MatrixClient): void {
     const cancelledKeyVerificationTxns: string[] = [];
     toDeviceMessages
-        .map(mapToDeviceEvent)
-        .map((toDeviceEvent) => {
+        .map((processedMessage) => {
             // map is a cheap inline forEach
             // We want to flag m.key.verification.start events as cancelled
             // if there's an accompanying m.key.verification.cancel event, so
             // we pull out the transaction IDs from the cancellation events
             // so we can flag the verification events as cancelled in the loop
             // below.
-            if (toDeviceEvent.getType() === "m.key.verification.cancel") {
-                const txnId: string = toDeviceEvent.getContent()["transaction_id"];
+            if (processedMessage.message.type === "m.key.verification.cancel") {
+                const txnId: string = processedMessage.message.content["transaction_id"];
                 if (txnId) {
                     cancelledKeyVerificationTxns.push(txnId);
                 }
             }
 
             // as mentioned above, .map is a cheap inline forEach, so return
             // the unmodified event.
-            return toDeviceEvent;
+            return processedMessage;
         })
-        .forEach(function (toDeviceEvent) {
-            const content = toDeviceEvent.getContent();
-            if (toDeviceEvent.getType() == "m.room.message" && content.msgtype == "m.bad.encrypted") {
-                // the mapper already logged a warning.
-                logger.log("Ignoring undecryptable to-device event from " + toDeviceEvent.getSender());
-                return;
-            }
-
-            if (
-                toDeviceEvent.getType() === "m.key.verification.start" ||
-                toDeviceEvent.getType() === "m.key.verification.request"
-            ) {
-                const txnId = content["transaction_id"];
-                if (cancelledKeyVerificationTxns.includes(txnId)) {
-                    toDeviceEvent.flagCancelled();
+        .forEach(function (processedEvent) {
+            // For backwards compatibility, we emit the event as a MatrixEvent
```

Comment:
Instead of emitting `processedEvent`? If so, it would help me if you added that to the comment.